src/integrationTest/java/uk/gov/hmcts/darts/common/controller/CourthouseApiTest.java

```diff
@@ -0,0 +1,168 @@
+            .contentType(MediaType.APPLICATION_JSON_VALUE);
+
+        MvcResult response = mockMvc.perform(requestBuilder).andExpect(status().isOk()).andDo(print()).andReturn();
+
+        List<ExtendedCourthouse> courthouseList = objectMapper.readValue(response.getResponse().getContentAsString(), new TypeReference<>() {});
+        ExtendedCourthouse haverfordwestCourthouse = objectMapper.readValue(haverfordwestResponse.getResponse().getContentAsString(), ExtendedCourthouse.class);
+        ExtendedCourthouse swanseaCourthouse = objectMapper.readValue(swanseaResponse.getResponse().getContentAsString(), ExtendedCourthouse.class);
+
+        assertTrue(courthouseList.contains(swanseaCourthouse));
+        assertTrue(courthouseList.contains(haverfordwestCourthouse));
+    }
+
+    /**
+     * Test adds courthouse and checks if returned object matches.
+     */
+    @Test
+    void courthousesPost() throws Exception {
+        MockHttpServletRequestBuilder requestBuilder = post("/courthouses")
+            .contentType(MediaType.APPLICATION_JSON_VALUE)
+            .content(getContentsFromFile(REQUEST_BODY_HAVERFORDWEST_JSON));
+        mockMvc.perform(requestBuilder).andExpect(status().isCreated())
+            .andExpect(jsonPath("$.courthouse_name", is("HAVERFORDWEST")))
+            .andExpect(jsonPath("$.code", is(761)))
+            .andExpect(jsonPath("$.created_date_time", is(notNullValue())))
+            .andExpect(jsonPath("$.last_modified_date_time", is(notNullValue())));
+    }
+
+    @Test
+    void courthousesPostMissingCourthouseName() throws Exception {
+        MockHttpServletRequestBuilder requestBuilder = post("/courthouses")
+            .contentType(MediaType.APPLICATION_JSON_VALUE)
+            .content(getContentsFromFile(REQUEST_BODY_400_MISSING_COURTHOUSE_NAME_JSON));
+        MvcResult response = mockMvc.perform(requestBuilder).andExpect(status().isBadRequest()).andReturn();
+        assertEquals("{\"code\":\"400 BAD_REQUEST\",\"message\":\"courthouseName must not be null\"}", response.getResponse().getContentAsString());
+    }
+
+    /**
+     * Test utility method used to add courthouse to database.
+     *
+     * @param fileLocation location of file that contains courthouse to be added.
+     * @return response for successful add
+     */
+    private MvcResult makeRequestToAddCourthouseToDatabase(String fileLocation) throws Exception {
+        MockHttpServletRequestBuilder requestBuilder = post("/courthouses")
+            .contentType(MediaType.APPLICATION_JSON_VALUE)
+            .content(getContentsFromFile(fileLocation));
+        return mockMvc.perform(requestBuilder).andExpect(status().is2xxSuccessful()).andDo(print()).andReturn();
+    }
+
+
+    /**
+     * Test adds courthouse then amends it and then retrieves it to check amendment was made.
+     */
+    @Test
+    void courthousesPut() throws Exception {
+        makeRequestToAddCourthouseToDatabase(
```

Comment:
there is no `deleteAll()` before or after any of the tests, so can they interfere with eachother?

Reply:
ACCEPT, updated tests as well.